File: autox/src/autox/log/Log.java
package autox.log;

public class Log {
	public static void debug(Object message){
		LoggerInstance.getInstance().get().debug(message);
	}
	public static void debug(Object message,Throwable t){
		LoggerInstance.getInstance().get().debug(message,t);
	}
	
	public static void info(Object message){
		LoggerInstance.getInstance().get().info(message);
	}
	public static void info(Object message,Throwable t){
		LoggerInstance.getInstance().get().info(message,t);
	}
	
	public static void warn(Object message){
		LoggerInstance.getInstance().get().warn(message);
	}
	public static void warn(Object message,Throwable t){
		LoggerInstance.getInstance().get().warn(message,t);
	}
	
	public static void error(Object message){
		LoggerInstance.getInstance().get().error(message);
	}
	public static void error(Object message,Throwable t){
		LoggerInstance.getInstance().get().error(message,t);
	}
	
	public static void fatal(Object message){
		LoggerInstance.getInstance().get().fatal(message);
	}
	public static void fatal(Object message,Throwable t){
		LoggerInstance.getInstance().get().fatal(message,t);
	}
}

File: dotnet/AutoX.Basic/Log.cs
#region

using System.Diagnostics;

#endregion

namespace AutoX.Basic
{
    public static class Log
    {
        public static void Debug(string message)
        {
            Logger.GetInstance().Log().Debug(GetPosition() + message);
        }

        public static void Info(string message)
        {
            Logger.GetInstance().Log().Info(GetPosition() + message);
        }

        public static void Warn(string message)
        {
            Logger.GetInstance().Log().Warn(GetPosition() + message);
        }

        public static void Error(string message)
        {
            Logger.GetInstance().Log().Error(GetPosition() + message);
        }

        public static void Fatal(string message)
        {
            Logger.GetInstance().Log().Fatal(GetPosition() + message);
        }

        private static string GetPosition()
        {
            var st = new StackTrace(new StackFrame(2, true));

            var currentFile = st.GetFrame(0).GetFileName();
            var currentLine = st.GetFrame(0).GetFileLineNumber();
            return "File:" + currentFile + " Line:" + currentLine + "\t";
        }
    }
}